src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -22,56 +21,83 @@
     }
 
+    /**
+     * Constructs a DeleteCommand for deletion by index
+     * @param targetIndex The index of the person to be deleted on the filtered list on GUI
+     */
+    public DeleteCommand(Index targetIndex) {
+        this.nameOfPersonToDelete = new Name("name"); //a dummy value
+        this.targetIndex = targetIndex;
+        this.isDeletionByIndex = true;
+    }
+
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
 
         requireNonNull(model);
 
-        Person personWithNameToDelete = new Person(nameOfPersonToDelete);
-
-        if (!model.hasPerson(personWithNameToDelete)) { // model.hasPerson considers 2 Persons with same name
-            // to be the same Person
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_NAME);
-        }
-
-        ObservableList<Person> personList = model.getAddressBook().getPersonList();
+        if (isDeletionByIndex) { //deletion by index
+            List<Person> lastShownList = model.getFilteredPersonList();
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+            Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+            model.deletePerson(personToDelete);
+            return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
 
-        final Person[] personToDelete = new Person[1];
+        } else { //deletion by name
+            Person personWithNameToDelete = new Person(nameOfPersonToDelete);
 
-        personList.forEach((person) -> {
-            if (person.isSamePerson(personWithNameToDelete)) {
-                personToDelete[0] = person;
-                model.deletePerson(person);
+            if (!model.hasPerson(personWithNameToDelete)) { //model.hasPerson considers 2 Persons with same name
+                //to be the same Person
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_NAME);
             }
-        });
-
-        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete[0]));
+            //todo : make this more oop
+            ObservableList<Person> personList = model.getAddressBook().getPersonList();
+            Person personToDelete = personWithNameToDelete; //dummy person
+
+            for (Person personFromList : personList) {
+                if (personFromList.isSamePerson(personWithNameToDelete)) {
+                    personToDelete = new Person(personFromList.getName(),
+                            personFromList.getPhone(), personFromList.getEmail(),
+                            personFromList.getAddress(), personFromList.getDescription(),
+                            personFromList.getTags(), personFromList.getLogs());
+                    model.deletePerson(personFromList);
+                    break;
+                }
```

Comment:
Maybe consider adding a method in UniqueFriendList called findIndex? It is good OOP and also it may be reuseable in other areas.

Reply:
Yup I agree! i think i would implement a method to get a person by name